test/junit/scala/tools/nsc/backend/jvm/BytecodeTest.scala

```diff
@@ -145,11 +145,12 @@ class BytecodeTest extends BytecodeTesting {
 
     // t8: no null checks invoking equals on modules and constants
     assertSameCode(getMethod(c, "t8"), List(
-      Field(GETSTATIC, "scala/collection/immutable/Nil$", "MODULE$", "Lscala/collection/immutable/Nil$;"), VarOp(ALOAD, 1), Invoke(INVOKEVIRTUAL, "java/lang/Object", "equals", "(Ljava/lang/Object;)Z", false), Jump(IFNE, Label(10)),
-      Ldc(LDC, ""), VarOp(ALOAD, 1), Invoke(INVOKEVIRTUAL, "java/lang/Object", "equals", "(Ljava/lang/Object;)Z", false), Jump(IFNE, Label(14)),
-      Label(10), Op(ICONST_1), Jump(GOTO, Label(17)),
-      Label(14), Op(ICONST_0),
-      Label(17), Op(IRETURN)))
+      Field(GETSTATIC, "scala/package$", "MODULE$", "Lscala/package$;"), Invoke(INVOKEVIRTUAL, "scala/package$", "Nil", "()Lscala/collection/immutable/Nil$;", false), VarOp(ALOAD, 1), VarOp(ASTORE, 2), Op(DUP), Jump(IFNONNULL, Label(12)),
+      Op(POP), VarOp(ALOAD, 2), Jump(IFNULL, Label(23)), Jump(GOTO, Label(17)),
```

Comment:
here as well, the point of the test is "no null checks invoking equals on modules", so let's change the definition of `t8` instead